implementation/src/main/java/io/smallrye/openapi/runtime/scanner/OpenApiAnnotationScanner.java

```diff
@@ -1872,6 +1875,21 @@ private Object parseExtensionValue(String value) {
         return value;
     }
 
+    
+    private CustomSchemaRegistry getCustomSchemaRegistry() {
+        if (config == null || config.customSchemaRegistryClass() == null) {
+            // Provide default implementation that does nothing
+            return (type) -> {};
+        } else {
+            try {
+                return (CustomSchemaRegistry) Class.forName(config.customSchemaRegistryClass()).newInstance();
```

Comment:
I think this is the only thing I'm a bit concerned about.  I think `Class.forName` isn't going to work well in all cases and we're going to need to be able to pass the classloader in (notice how it's done for filter and modelreader).  We'll probably need to pass the CL in similarly.

Reply:
OK, will check it

Reply:
Context classloader is much better, for sure.  I'm inclined to merge this and if consumers of this library need something more fine grained, we can address it later.